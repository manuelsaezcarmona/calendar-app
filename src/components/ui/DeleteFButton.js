/* eslint-disable react/button-has-type */
import React from 'react';
import { useDispatch } from 'react-redux';
import { eventDeleted } from '../../redux/actioncreators/event.actioncreator';

export function DeleteFButton() {
  const dispatch = useDispatch();

  const handleDelete = () => {
    dispatch(eventDeleted());
  };

  return (
    <button className="btn btn-danger fab-danger" onClick={handleDelete}>
      <i className="fas fa-trash" />
      <span> Borrar Evento</span>
    </button>
  );
}
