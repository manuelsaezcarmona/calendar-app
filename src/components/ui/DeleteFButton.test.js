/** El test no es tan complicado como la preparacion del entorno para ejecutar el test */
import React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
// esto es para tener la ayuda de los metodos de jest
import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { startEventDelete } from '../../redux/actioncreators/event.actioncreator';
import { DeleteFButton } from './DeleteFButton';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {};

const store = mockStore(initState);
// En el componente disparo una accion (asincrona) que ya testee en las actions.
// voy a hacer un mock de esa accion.
store.dispatch = jest.fn();

const wrapper = mount(
  <Provider store={store}>
    <DeleteFButton />
  </Provider>
);

describe('Given the DeleteFbutton', () => {
  test('Shuuld renderizer correctry', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
