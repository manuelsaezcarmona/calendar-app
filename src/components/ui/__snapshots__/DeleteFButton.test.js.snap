// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the DeleteFbutton Shuuld renderizer correctry 1`] = `
<button
  className="btn btn-danger fab-danger"
  onClick={[Function]}
>
  <i
    className="fas fa-trash"
  />
  <span>
     Borrar Evento
  </span>
</button>
`;
